core/arch/arm/kernel/generic_entry_a32.S

```diff
@@ -568,15 +568,18 @@ UNWIND(	.cantunwind)
 	cpu_is_ready
 
 #if defined (CFG_BOOT_SECONDARY_REQUEST)
-	/* generic_boot_core_hpen return value (r0) is ns entry point */
+	/* generic_boot_core_hpen return value (r0) is ns entry context structure */
 	bl	generic_boot_core_hpen
+	ldr	r6, [r0, #4]	/* load nonsecure r0 value to r6 */
+	ldr	r0, [r0, #0]	/* load entry point address to r0 */
```

Comment:
Could you use some `#define NSENTRY_EP_ADDR_OFFS 0` and `#define NSENTRY_CONTEXT_ID_OFFS 4`? it defines the magic numbers and prevents need for a comment.